pkg/sql/schemachanger/sctest/cumulative.go

```diff
@@ -52,34 +481,66 @@ import (
 func cumulativeTest(
 	t *testing.T,
 	relPath string,
-	tf func(t *testing.T, path string, rewrite bool, setup, stmts []parser.Statement),
+	tf func(t *testing.T, path string, rewrite bool, setup, stmts []parser.Statement, stageExecMap *stageExecStmtMap),
 ) {
 	skip.UnderStress(t)
 	skip.UnderRace(t)
 	path := testutils.RewritableDataPath(t, relPath)
 	var setup []parser.Statement
-	datadriven.RunTest(t, path, func(t *testing.T, d *datadriven.TestData) string {
-		stmts, err := parser.Parse(d.Input)
-		require.NoError(t, err)
-		require.NotEmpty(t, stmts)
+	stageExecMap := makeStageExecStmtMap()
+	rewrite := false
+	var testStmts parser.Statements
+	var lines []string
+	numTestStmts := 0
 
+	// First pass collect information.
```

Comment:
My understanding of the current implementation is this: We scan the testdata file from top to bottom. We require a command ordering of `setup` --> `stage-exec`/`stage-query` --> `test`.
1. for `setup`: nothing changes, run the set up commands;
2. for `stage-exec`/`stage-query`: remember them in som data structure;
3. for `test`: run the actual test command, with the recorded (from step 2) stmts injected.

Is this understanding correct? If so, maybe this comment line "First pass collect information" isn't very accurate bc we not only just collect information -- we indeed executed all the statements and did all the injection.

Reply:
Changed this to describe it better
```
        // First pass collect stage-exec/stage-query/setup commands and execute them
	// test once the test command is encountered. Only a single test command is
	// allowed via an assertion which guarantees all others appear first.
```